### PR TITLE
perf: remove match where possible

### PR DESCRIPTION
**Pull request by amaanq**
Bound to have made a silly mistake somewhere :grin:

### CODE REVIEW

**Review by clason — APPROVED** [review]

File: queries/t32/highlights.scm [hunk 1]
```diff
@@ -97,14 +97,14 @@
...
- (#match? @constant.builtin "^[%/][a-zA-Z][a-zA-Z0-9.]*$")
+ (#lua-match? @constant.builtin "^[%%/][%l%u][%l%u%d.]*$")
 )
 
 (
   (command_expression
     command: (identifier) @keyword
     arguments: (argument_list . (identifier) @label))
-  (#match? @keyword "^[gG][oO][tT][oO]$")
+  (#lua-match? @keyword "^[gG][oO][tT][oO]$")
```
Comment (ObserverOfTime) [inline queries/t32/highlights.scm:107]:
```suggestion
  (#any-of? @keyword "goto" "GOTO")
```

**Review by ObserverOfTime — CHANGES_REQUESTED** [review]

File: queries/t32/highlights.scm (hunk 1 above)
Comment (clason) [reply queries/t32/highlights.scm:107]: Won't match `Goto`.

File: queries/dockerfile/highlights.scm [hunk 2]
```diff
@@ -60,4 +60,4 @@
   (#match? @number "[0-9][0-9]?$"))
 
 ((stopsignal_instruction) @constant.builtin
-  (#match? @constant.builtin "SIG(ABRT|HUP|INT|KILL|QUIT|STOP|TERM|TSTP)$"))
```
[the remark this answers is not in the document]
Comment (amaanq) [reply queries/dockerfile/highlights.scm:63]: Hmm, what do you mean? That the rule can never be captured to begin w/?

File: queries/smali/highlights.scm [hunk 3]
```diff
@@ -82,7 +82,10 @@
   (#lua-match? @keyword.return "^return"))
 
 ((opcode) @conditional
-  (#match? @conditional "^(if|cmp)"))
+  (#lua-match? @conditional "^cmp"))
```
Comment (lucario387) [inline queries/smali/highlights.scm:85]: Missing if here

File: queries/starlark/highlights.scm [hunk 4]
```diff
@@ -29,7 +29,7 @@
 
 ((attribute
     attribute: (identifier) @field)
- (#match? @field "^([A-Z])@!.*$"))
+ (#lua-match? @field "^%d@!.*$"))
```
Comment (lucario387) [inline queries/starlark/highlights.scm:32]: is this %u or %d

File: queries/t32/highlights.scm [hunk 5]
```diff
@@ -76,12 +76,12 @@
 (
   (command_expression
     command: (identifier) @keyword.return)
-  (#match? @keyword.return "^[eE][nN][dD]([dD][oO])?$")
+  (#any-of? @keyword.return "end" "END" "enddo" "ENDDO")
 )
 (
   (command_expression
     command: (identifier) @keyword.return)
-  (#match? @keyword.return "^[rR][eE][tT][uU][rR][nN]$")
```
Comment (lucario387) [inline queries/t32/highlights.scm:84]: Do languages like this allow for things like `rEtUrN` or not?

Comment (ObserverOfTime) [reply queries/t32/highlights.scm:84]: Presumably. But would anyone write it like that?

File: queries/dockerfile/highlights.scm (hunk 2 above)
[the remark this answers is not in the document]
Comment (lucario387) [reply queries/dockerfile/highlights.scm:63]: Probably some case like this? 
https://github.com/nvim-treesitter/nvim-treesitter/pull/4430/files

Where the signal is something else under the node and not it exactly

File: queries/t32/highlights.scm (hunk 5 above)
Comment (clason) [reply queries/t32/highlights.scm:84]: If the grammar allows it, it should be highlighted correctly.

File: queries/dockerfile/highlights.scm (hunk 2 above)
[the remark this answers is not in the document]
Comment (ObserverOfTime) [reply queries/dockerfile/highlights.scm:63]: I think when I last checked the signal didn't have its own node and only the `STOPSIGNAL` keyword did.

[the remark this answers is not in the document]
Comment (ObserverOfTime) [reply queries/dockerfile/highlights.scm:63]: I think when I last checked the signal didn't have its own node and only the `STOPSIGNAL` keyword did.

File: queries/t32/highlights.scm [hunk 6]
```diff
@@ -97,14 +97,14 @@
...
- (#match? @constant.builtin "^[%/][a-zA-Z][a-zA-Z0-9.]*$")
+ (#lua-match? @constant.builtin "^[%%/][%l%u][%l%u%d.]*$")
 )
 
 (
   (command_expression
     command: (identifier) @keyword
     arguments: (argument_list . (identifier) @label))
-  (#match? @keyword "^[gG][oO][tT][oO]$")
+  (#lua-match? @keyword "[gG][oO][tT][oO]]")
```
Comment (ObserverOfTime) [inline queries/t32/highlights.scm:107]:
```suggestion
  (#lua-match? @keyword "^[gG][oO][tT][oO]$")
```

File: queries/dockerfile/highlights.scm (hunk 2 above)
[the remark this answers is not in the document]
Comment (ObserverOfTime) [reply queries/dockerfile/highlights.scm:63]: `. _` won't work either when there is no node.

[the remark this answers is not in the document]
Comment (ObserverOfTime) [reply queries/dockerfile/highlights.scm:63]: `. _` won't work either when there is no node.

File: queries/smali/highlights.scm [hunk 7]
```diff
@@ -82,7 +82,13 @@
...
 
 ((opcode) @conditional
-  (#match? @conditional "^(if|cmp)"))
+  (#lua-match? @conditional "^if"))
+
+((opcode) @conditional
+  (#lua-match? @conditional "^cmp"))
+
+((opcode) @exception
+  (#lua-match? @exception "^throw"))
```
Comment (ObserverOfTime) [inline queries/smali/highlights.scm:91]: Duplicate?
```suggestion
```

File: queries/dockerfile/highlights.scm (hunk 2 above)
[the remark this answers is not in the document]
Comment (amaanq) [reply queries/dockerfile/highlights.scm:63]: Hmm does it not? I was thinking _ would capture anything after "STOPSIGNAL" here: https://github.com/camdencheek/tree-sitter-dockerfile/blob/8ee3a0f7587b2bd8c45c8cb7d28bd414604aec62/grammar.js#L155

File: queries/t32/highlights.scm [hunk 8]
```diff
@@ -97,14 +97,14 @@
...
- (#match? @constant.builtin "^[%/][a-zA-Z][a-zA-Z0-9.]*$")
+ (#lua-match? @constant.builtin "^[%%/][%l%u][%l%u%d.]*$")
 )
 
 (
   (command_expression
     command: (identifier) @keyword
     arguments: (argument_list . (identifier) @label))
-  (#match? @keyword "^[gG][oO][tT][oO]$")
+  (#lua-match? @keyword "[gG][oO][tT][oO]")
```
Comment (ObserverOfTime) [inline queries/t32/highlights.scm:107]: It will also match `foogotobar` without `^ $`.

Comment (amaanq) [reply queries/t32/highlights.scm:107]: whoops

File: queries/dockerfile/highlights.scm (hunk 2 above)
[the remark this answers is not in the document]
Comment (ObserverOfTime) [reply queries/dockerfile/highlights.scm:63]: > If not then I'll just remove it, the grammar should be updated to allow users to capture that node

Do the same for the capture above it.

File: queries/smali/highlights.scm (hunk 7 above)
Comment (amaanq) [reply queries/smali/highlights.scm]: aha, I copied & pasted w/o replacing it w/ if

**Comment by amaanq** [comment]
Can't merge even though I resolved your suggested changes @ObserverOfTime, mind checking that out?

**Review by ObserverOfTime — APPROVED** [review]

**Pull request merged**